File: device.c
#include "vsm_api.h"

enum
{
	A, B, C
} PINS;

VSM_PIN device_pins[]=
{
	{.is_digital=TRUE, .name = "A", .number = 1, .on_time=15000000000, .off_time=22000000000},
	{.is_digital=TRUE, .name = "B", .number = 2, .on_time=15000000000, .off_time=22000000000},
	{.is_digital=TRUE, .name = "C", .number = 3, .on_time=15000000000, .off_time=22000000000},
	{.name = 0}
};

void device_simulate ( void )
{
	
}

File: src/device/device.lua
-- Here you can place you device code. 
-- It will be automatically compiled at build time
-- Comment out this to use predefined script
--[[
__USE_PRECOMPILED = 0xDEADC0DE

--NAND gate placeholder script
SAFE_MODE=true
LOGIC_TYPE=TTL

device_pins =
{
    {name = "TX", on_time=100000, off_time=100000} 
}

function device_simulate ()
    Q:set(1-(A:get() * B:get()))
end
]]--